runtime/stream.c

```diff
@@ -480,7 +480,13 @@ strmNextFile(strm_t *pThis)
 	 * make sure their iMaxFiles is large enough. But it is well-desired for other
 	 * use cases, e.g. a circular output log file. -- rgerhards, 2008-01-10
 	 */
-	pThis->iCurrFNum = (pThis->iCurrFNum + 1) % pThis->iMaxFiles;
+	if((pThis->iCurrFNum + 1) == 0 || pThis->iMaxFiles == 0) {
```

Comment:
So on line 472 there is an assertion that `pThis->iMaxFiles` is not zero, so why do we still need this check in the conditional?

Without the need for checking for `pThis->iMaxFiles` is zero, won't that just revert back to what was originally there?

That is to say, we only need to change the sign of the variables used, right?